### PR TITLE
Feature/firebaseanalytics

### PR DESCRIPTION
**Pull request by NunoSSilva**

### CODE REVIEW

File: adal-analytics/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -28,6 +28,7 @@
 
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
     <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.WAKE_LOCK"/>
```
Comment (jzeferino) [inline adal-analytics/src/main/AndroidManifest.xml:31]: Why is this needed @NunoSSilva ?

File: adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a screenName to send the Screen");
+            return;
+        }
+        sendScreen(activity, activity.getString(screenNameId));
+    }
+
+    /**
+     * Send a screen with screenId
+     *
+     * @param activity     the activity the activity
```
Comment (jzeferino) [inline adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java]: Bad description.

File: adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+     */
+    public static void sendScreen(Activity activity, String screenName) {
+
+        if (TextUtils.isEmpty(screenName)) {
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a screenName to send the Screen");
+            return;
+        }
+
+        if (BuildConfig.DEBUG) {
+            Log.i(FirebaseAnalyticsManager.class.getCanonicalName(), "Setting Screen name: " + screenName);
```
Comment (jzeferino) [inline adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java]: This could be `Log.d`.

File: adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    /**
+     * Send an event with eventName with all event types on bundle
+     *
+     * @param activity  the activity
+     * @param eventName name that will appear on timeline
+     * @param eventInfo info about the event
+     */
+    public static void sendEvent(Activity activity, String eventName, Map<String, String> eventInfo) {
+        if (TextUtils.isEmpty(eventName)) {
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a eventName to send the Event");
```
Comment (jzeferino) [inline adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java:122]: This `if` is duplicated in other methods. Should be placed in a separated method.

File: adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+            return;
+        }
+        if (eventInfo == null || eventInfo.isEmpty()) {
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a eventInfo to send the Event");
+            return;
+        }
+
+        Bundle bundle = new Bundle();
+        Set<Map.Entry<String, String>> setEntry = eventInfo.entrySet();
+        for (Map.Entry<String, String> entry : setEntry) {
```
Comment (jzeferino) [inline adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java]: Could be used `eventInfo.entrySet()` in the `foreach` loop directly instead of declaring it.

File: adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+    /**
+     * Send a user property (propertyId) with value
+     *
+     * @param activity   the activity
+     * @param propertyId resource id to property string
+     * @param value      the value
+     */
+    public static void sendUserProperty(Activity activity, @StringRes int propertyId, String value) {
+        if (propertyId == 0) {
+            Log.w(FirebaseAnalyticsManager.class.getCanonicalName(), "You need a propertyId to send the User Property");
```
Comment (jzeferino) [inline adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java:162]: More repeated code.

File: adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+    /**
+     * Send a user property (property) with value
+     *
+     * @param activity the activity
+     * @param property the property name
+     * @param value    the value
+     */
+    public static void sendUserProperty(Activity activity, String property, String value) {
+        if (TextUtils.isEmpty(property)) {
```
Comment (jzeferino) [inline adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java:176]: More repeated code.

File: gradle.properties [hunk 8]
```diff
@@ -25,6 +25,7 @@
 org.gradle.jvmargs=-Xmx1536M
 libraryVersionName=0.0.0
 supportVersion=25.3.1
-googlePlayServicesVersion=10.2.6
+googlePlayServicesVersion=11.0.1
```
Comment (jzeferino) [inline gradle.properties:28]: 👍

File: adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java (hunk 2 above)
Comment (NunoSSilva) [reply adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java]: What is your suggestion?

Comment (jzeferino) [reply adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java]: `* @param activity     the activity`

File: adal-analytics/src/main/AndroidManifest.xml (hunk 1 above)
Comment (NunoSSilva) [reply adal-analytics/src/main/AndroidManifest.xml:31]: Because when you call the FirebaseAnalytics.getInstance() the IDE shows a warning with: "Missing permissions required by FirebaseAnalytics.getInstance: android.permission.WAKE_LOCK"

File: adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java (hunk 3 above)
Comment (NunoSSilva) [reply adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java]: 👍

File: adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java (hunk 5 above)
Comment (NunoSSilva) [reply adal-analytics/src/main/java/com/massivedisaster/adal/analytics/FirebaseAnalyticsManager.java]: 👍

**Review by jzeferino — APPROVED** [review]
